sunpy/net/dataretriever/sources/gong.py

```diff
@@ -90,22 +91,28 @@ def _get_url_for_timerange(self, timerange, **kwargs):
                 patterns.append(url_pattern_1)
             else:
                 #Differentiate on basis of wavelength
+                
+                #The isclose function in numpy can only check two arrays if they
+                #are close to each other within some precision. Standalone values such
+                #as int, doubles etc can't be checked that way. In order to do that, we put the
+                #two indiviual values in two seperate arrays da and db and then apply
+                #isclose on both those arrays.
                 if not wavelength_in:
                     patterns.append(url_pattern_1), patterns.append(url_pattern_2)
                 else:
                     try:
                         wave_float = kwargs['wavelength'].min.value
-                        da = []
+                        da = list()
                         da.append(wave_float)
                         for wave_nums in pattern_table.keys():
-                            db = []
+                            db = list()
                             db.append(wave_nums)
                             if np.isclose(da, db, 1e-10, 1e-10):
                                 wave = wave_nums
                                 break
                         patterns.append(pattern_table[wave])
-                    except NameError:
-                        print ("Enter correct wavelength range and units")
+                    except:
+                        raise NameError("Enter correct wavelength range and units")
```

Comment:
before were just raising if a nameeror happens. Now you are just raising a nameerror in any circunstance. Be careful with these things when coding. Imagine that you had a typo in isclose, or in a variable... and you will suffer to find it.